crates/wadm/src/scaler/secretscaler.rs

```diff
@@ -0,0 +1,119 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use std::hash::{DefaultHasher, Hash, Hasher};
+use tokio::sync::RwLock;
+use tracing::{debug, error, instrument, trace};
+use wadm_types::{
+    api::{StatusInfo, StatusType},
+    SecretSourceProperty, TraitProperty,
+};
+
+use crate::{
+    commands::{Command, DeleteConfig, PutConfig},
+    events::{ConfigDeleted, ConfigSet, Event},
+    scaler::Scaler,
+    workers::SecretSource,
+};
+
+const SECRET_PREFIX: &str = "secret";
+
+pub struct SecretScaler<SecretSource> {
+    secret_source: SecretSource,
+    id: String,
+    secret_name: String,
+    source: SecretSourceProperty,
+    status: RwLock<StatusInfo>,
+}
+
+impl<S: SecretSource> SecretScaler<S> {
+    pub fn new(secret_name: String, source: SecretSourceProperty, secret_source: S) -> Self {
+        let mut id = secret_name.clone();
+        let mut hasher = DefaultHasher::new();
+        source.hash(&mut hasher);
+        id.extend(format!("-{}", hasher.finish()).chars());
+
+        Self {
+            id,
+            secret_name,
+            source,
+            secret_source,
+            status: RwLock::new(StatusInfo::reconciling("")),
+        }
+    }
+}
+
+#[async_trait]
+impl<S: SecretSource + Send + Sync + Clone> Scaler for SecretScaler<S> {
+    fn id(&self) -> &str {
+        todo!()
```

Comment:
:scream:

since this is a uniqueness check, combination of secret name + source should be enough?

Reply:
This was just missing the implementation of returning `&self.id`. Just something I didn't get a chance to fill in! The id uses the same hashing logic that the configscaler uses